### ISSUE TITLE
idf.open_subdomains: give more informative error for failures

### ISSUE THREAD

**Issue by Manangka** [Deltares/imod-python#75]
In GitLab by @T.Mulder on Jun 28, 2019, 11:12

I have a case where I'm reading in 3 subdomains. However, there's another result with another subdomain. In this case, the dask array merge fails because the created arrays do not have matching shapes.

Should be easy to check with looking at the number of paths:
```python
n_subdomains_members = [len(member) for member in grouped.values()]
if not all(n_subdomains_members == n_subdomain_members[0]):
    raise ValueError("Unequal numbers of IDFs detected per subdomain number")
```

**Comment by Manangka** [comment]
In GitLab by @Huite on Jun 28, 2019, 13:07

Another thing we noticed:
xr.merge returns a dataset instead of a DataArray.

And interestingly, loading the IDF subdomains, and then writing the IDFs is incredibly slow.
I'm thinking the chunks are getting merged one by one; it's streaming the data, one set of IDFs at a time. That means you get a merge operation for every layer, and for every time.

A work-around is calling .load().
If it doesn't fit into memory completely, I think one option would be to rechunk to larger sizes (multiple IDFs).
I think the ultimate cause is a failure in the scheduling mechanism of xr.merge. We've already seen that stacking dask arrays is much faster, but it's rather a headache for subdomains.

**Comment by Manangka** [comment]
In GitLab by @Huite on Jun 28, 2019, 13:53

Fixed in 9e55f557bb571e40f33f76602e6ea81415063d1e

**Comment by Manangka** [comment]
In GitLab by @Huite on Jun 28, 2019, 13:53

closed
